### PR TITLE
Use proper section macros in networking

### PR DESCRIPTION
**Pull request by jukkar**
Convert networking code to use `Z_STRUCT_SECTION_FOREACH()` and `Z_STRUCT_SECTION_ITERABLE()` macros.
Also remove special alignment from net_if and ppp_protocol_handlers structs as they no longer seem to be needed.

### CODE REVIEW

File: include/linker/common-rom.ld [hunk 1]
```diff
@@ -109,12 +101,11 @@
 #endif
 
 #if defined(CONFIG_NET_L2_PPP)
-	SECTION_PROLOGUE(net_ppp_proto,,)
+	SECTION_PROLOGUE(_ppp_protocol_handler,,)
```
Comment (jenmwms) [inline include/linker/common-rom.ld:104]: Can you please describe briefly why this needs to change to _protocol_handler?

**Comment by jenmwms** [comment]
> as they no longer seem to be needed.

Can you please share what brings you to this conclusion?

File: include/linker/common-rom.ld (hunk 1 above)
Comment (jukkar) [reply include/linker/common-rom.ld:104]: The system did not crash in short testing (more needed with different hw to fully verify). Also the said macro will have Z_DECL_ALIGN() call which I hope will set the proper alignment automatically.

**Review by andyross — APPROVED** [review]

**Review by jenmwms — APPROVED** [review]

**Comment by carlescufi** [comment]
@jukkar please rebase

**Review by tyrantlizard77 — COMMENTED** [review]
@jukkar can you rebase this and use Z_ITERABLE_SECTION_ROM/Z_ITERABLE_SECTION_RAM in the linker scripts?

**Comment by carlescufi** [comment]
> @jukkar can you rebase this and use Z_ITERABLE_SECTION_ROM/Z_ITERABLE_SECTION_RAM in the linker scripts?

@andrewboie this is now done. Could you take another look?

File: include/linker/common-ram.ld [hunk 2]
```diff
@@ -99,29 +99,11 @@
...
-	{
-		__net_l2_data_start = .;
-		*(".net_l2.data")
-		KEEP(*(SORT_BY_NAME(".net_l2.data*")))
-		__net_l2_data_end = .;
-	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
+#if defined(CONFIG_NETWORKING)
+	Z_ITERABLE_SECTION_RAM(net_if, 4)
+	Z_ITERABLE_SECTION_RAM(net_if_dev, 4)
+	Z_ITERABLE_SECTION_RAM(net_l2, 4)
```
Comment (jukkar) [inline include/linker/common-ram.ld:105]: Hmm, this should be called net_l2_data, I will resend a new version.

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**